### PR TITLE
🐛Fix amp-smartlinks incorrectly formatted edit URLs

### PR DESCRIPTION
**Pull request by nickcroce**
### Short description of what this resolves:
We were running into an issue where edit URLs were being ingested from our extension in the following format: 
`https://www-example-com.cdn.ampproject.org/v/s/www.example.com/example/best-example`
_Note that the actual base URL is being appended to the end of the example above_

We decided the best approach to fixing this was from the source, rather than down the pipeline.

### Changes proposed in this pull request:
Fix the faulty URLs by extracting the actual URL base prior to compiling edit data.

### CODE REVIEW

File: extensions/amp-smartlinks/0.1/linkmate.js [hunk 1]
```diff
@@ -163,12 +163,34 @@ export class Linkmate {
...
+   * @return {?string}
+   * @private
+   */
+  extractBaseUrl_(currentUrl) {
+    const splitPath = currentUrl.split('/');
+    let startIndex = 0;
+    while (startIndex < splitPath.length) {
+      if (splitPath[startIndex].startsWith('www.') ||
+        splitPath[startIndex].endsWith('.com')) {
+        return splitPath.slice(startIndex).join('/');
```
Comment (Abalaxy) [inline extensions/amp-smartlinks/0.1/linkmate.js]: We'll lose protocol here, right? I think we'll need that for urls to match from our normal jstag. ex: https://narrativ.looker.com/sql/jgq7bxwbjshzwk and https://narrativ.looker.com/sql/ftbhmxdjrh59vp, but not https://narrativ.looker.com/sql/m9zbfdngrvzcc3

File: extensions/amp-smartlinks/0.1/linkmate.js [hunk 2]
```diff
@@ -163,12 +163,34 @@ export class Linkmate {
...
   }
 
+  /**
+   * Truncates AMP URLs to extract the base url for Narrativ Edit Tracking
+   * Purposes
+   * @param {?string} currentUrl The current URL of the AMP page
+   * @return {?string}
+   * @private
+   */
+  extractBaseUrl_(currentUrl) {
```
Comment (PhilWinchester) [inline extensions/amp-smartlinks/0.1/linkmate.js:190]: couldn't we make this a bit more naive and split on `www.` and then just re-assemble the url from there? saves us the while loop.

File: extensions/amp-smartlinks/0.1/linkmate.js (hunk 1 above)
Comment (nickcroce) [reply extensions/amp-smartlinks/0.1/linkmate.js]: True. I'll grab protocol and prepend

File: extensions/amp-smartlinks/0.1/linkmate.js (hunk 2 above)
Comment (nickcroce) [reply extensions/amp-smartlinks/0.1/linkmate.js:190]: Doy! Good call

Comment (myfashionhub) [reply extensions/amp-smartlinks/0.1/linkmate.js:190]: Could also try using regex. "while" seems a bit inefficient 😅

File: extensions/amp-smartlinks/0.1/test/test-linkmate.js [hunk 3]
```diff
@@ -415,6 +415,36 @@ describes.fakeWin(
...
+            .returns('https://www-example-com.cdn.ampproject.org/v/s/' +
+              'www.example.com/example/best-example');
+        env.sandbox.spy(linkmate, 'getEditInfo_');
+
+        const expectedPayload = {
+          'name': 'Fake Website Title',
+          'url': 'www.example.com/example/best-example',
+        };
+
+        const editPayload = linkmate.getEditInfo_();
```
Comment (myfashionhub) [inline extensions/amp-smartlinks/0.1/test/test-linkmate.js]: could write this inline

File: extensions/amp-smartlinks/0.1/linkmate.js (hunk 1 above)
Comment (nickcroce) [reply extensions/amp-smartlinks/0.1/linkmate.js]: @Abalaxy @PhilWinchester check er out, what do you think?

Comment (nickcroce) [reply extensions/amp-smartlinks/0.1/linkmate.js]: Added a conditional to check for `www-` as well so we don't unnecessarily split up the url

**Review by Abalaxy — APPROVED** [review]
LGTM, pending Phil's review

File: extensions/amp-smartlinks/0.1/linkmate.js [hunk 4]
```diff
@@ -163,12 +163,30 @@ export class Linkmate {
    * @private
    */
   getEditInfo_() {
+    let currentUrl = this.ampDoc_.getUrl();
+    if (currentUrl.includes('www-')) {
```
Comment (PhilWinchester) [inline extensions/amp-smartlinks/0.1/linkmate.js]: Would a regex for `https://www.` where the s is optional be more thorough?

**Review by PhilWinchester — COMMENTED** [review]
LGTM as is. 1 thing to think about for AMP is that highly specific code is hard to change quickly b/c of the deployment cycle. Is there a more generic way you could create a url extractor that would work on a larger set of junky urls?

File: extensions/amp-smartlinks/0.1/linkmate.js (hunk 4 above)
Comment (nickcroce) [reply extensions/amp-smartlinks/0.1/linkmate.js]: Kind of tough to say considering we've only seen this specific example. Removing the conditional would probably be more thorough, just extracting the base of every url to be sure.

**Comment by nickcroce** [comment]
Closing this puppy - looks like google fixed the issue for us!

**Pull request closed without merging**